apps/src/blockly/addons/cdoTrashcan.js

```diff
@@ -2,6 +2,52 @@ import GoogleBlockly from 'blockly/core';
 import {ToolboxType} from '../constants';
 
 export default class CdoTrashcan extends GoogleBlockly.Trashcan {
+  /**
+   * @override
+   */
+  init() {
+    super.init();
+    this.workspace_.addChangeListener(this.workspaceChangeHandler.bind(this));
+  }
+
+  workspaceChangeHandler(blocklyEvent) {
+    if (blocklyEvent.type === Blockly.Events.BLOCK_DRAG) {
+      let trashcanDisplay = 'none';
+      let toolboxVisibility = 'visible';
```

Comment:
Does `toolboxVisibility` determine if the toolbox panel is expanded? or is it used for something else?

Reply:
it indicates whether the toolbox contents will be hidden or visible. when a block is dragged, the trashcan shows up over the toolbox area and we hide the toolbox contents.

Reply:
oh got it, so `trashcanDisplay` and `toolboxVisiblity` are kinda dependent on each other? would we ever have a scenario where we would want the toolbox to be visible and for the trashcan to show?

Reply:
Right, yeah, they should always be opposite- when the trashcan is shown the toolbox contents are hidden, and vice versa. If they both showed at once it'd look like this, which would not be desirable:
![image](https://user-images.githubusercontent.com/8787187/144330605-d008fa03-7292-448b-b228-ed42a3515bd6.png)

Reply:
ah I see! Thank you for breaking that down!!